clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -106,7 +111,7 @@
     private static final String LINGER_MS_DOC = "The producer groups together any records that arrive in between request transmissions into a single batched request. "
                                                 + "Normally this occurs only under load when records arrive faster than they can be sent out. However in some circumstances the client may want to "
                                                 + "reduce the number of requests even under moderate load. This setting accomplishes this by adding a small amount "
-                                                + "of artificial delay&mdash;that is, rather than immediately sending out a record the producer will wait for up to "
+                                                + "of artificial delay&mdash;that is, rather than immediately sending out a record, the producer will wait for up to "
```

Comment:
add a comma for better readability:
before:
`that is, rather than immediately sending out a record the producer will wait for up to the given delay to allow other records to be sent so that the sends can be batched together.`
after:
`that is, rather than immediately sending out a record[,] the producer will wait for up to the given delay to allow other records to be sent so that the sends can be batched together.`